src/ucp/rma/basic_rma.c

```diff
@@ -306,7 +306,19 @@ ucs_status_t ucp_get_nbi(ucp_ep_h ep, void *buffer, size_t length,
 
 ucs_status_t ucp_worker_fence(ucp_worker_h worker)
 {
-    return UCS_ERR_UNSUPPORTED;
+    unsigned rsc_index;
+
+    for (rsc_index = 0; rsc_index < worker->context->num_tls; ++rsc_index) {
+        if (worker->ifaces[rsc_index] == NULL) {
+            continue;
+        }
+
+        while (uct_iface_fence(worker->ifaces[rsc_index], 0) != UCS_OK) {
```

Comment:
@xinzhao3 ucp_worker_fence() should not return UCS_ERR_NO_RESOURCE, in case there are no resources it should put a fence request on pending queue. In case if any other error, it should fail.